### ISSUE TITLE
Concatenating issue when plotting analytic_pearson_residuals layers

### ISSUE THREAD

**Issue by ngvananh2508** [theislab/single-cell-tutorial#113]
when I ran these code which followed Single-cell best practices:

analytic_pearson = sc.experimental.pp.normalize_pearson_residuals(adata, inplace = False)
adata.layers["analytic_pearson_residuals"] = csr_matrix(analytic_pearson["X"])
p2 = sns.histplot(adata.layers["analytic_pearson_residuals"].sum(1), bins = 100, kde = False, ax = axes[1])

it appears this error: ValueError: No objects to concatenate.

I am running seaborn version 0.12.2 and scanpy version 1.9.3

Can anyone please help me with this issue?

Thank you so much!

**Comment by ngvananh2508** [comment]
I am running seaborn version 0.12.2 and scanpy version 1.9.3

**Comment by lconan** [comment]
I just ran into this error. I think it's on `sc.experimental.pp.normalize_pearson_residuals` where nan's are generated because of divided by 0 issue, at least in my case, and I don't know how to solve this

**Comment by ngvananh2508** [comment]
> I just ran into this error. I think it's on `sc.experimental.pp.normalize_pearson_residuals` where nan's are generated because of divided by 0 issue, at least in my case, and I don't know how to solve this

Do you get the kernel death problem when loading adata to robjects on jupyter lab in the feature selection part?

**Comment by lconan** [comment]
> > I just ran into this error. I think it's on `sc.experimental.pp.normalize_pearson_residuals` where nan's are generated because of divided by 0 issue, at least in my case, and I don't know how to solve this
> 
> Do you get the kernel death problem when loading adata to robjects on jupyter lab in the feature selection part?

No, I didn't. But I did get: `ValueError: Unknown dtype dtype('int64') cannot be converted to ?gRMatrix.`

**Comment by ngvananh2508** [comment]
> 



> > > I just ran into this error. I think it's on `sc.experimental.pp.normalize_pearson_residuals` where nan's are generated because of divided by 0 issue, at least in my case, and I don't know how to solve this
> > 
> > 
> > Do you get the kernel death problem when loading adata to robjects on jupyter lab in the feature selection part?
> 
> No, I didn't. But I did get: `ValueError: Unknown dtype dtype('int64') cannot be converted to ?gRMatrix.`

Are you practicing with single-cell best practice? May I keep contact with you in case if we have bugs, we can assist each other?

**Comment by ngvananh2508** [comment]
I solved all problems by checking the right file to run.

**Comment by lconan** [comment]
> I solved all problems by checking the right file to run.

I managed to solve this by converting the original count back to dense matrix and pass only .X to R functions.....I can't do this on test samples, because I do this on published dataset of others.

**Comment by ngvananh2508** [comment]
When I did like you on test sample, there was not any plot for this data.
